bioimageio/core/cli.py

```diff
@@ -0,0 +1,425 @@
+        return {}
+
+    req_dataset_meas, _ = get_required_dataset_measures(model_descr)
+    stat_adapter = TypeAdapter(
+        Mapping[DatasetMeasure, MeasureValue],
+        config=ConfigDict(arbitrary_types_allowed=True),
+    )
+
+    if stats_path.exists():
+        logger.info(f"loading precomputed dataset measures from {stats_path}")
+        stat = stat_adapter.validate_json(stats_path.read_bytes())
+        for m in req_dataset_meas:
+            if m not in stat:
+                raise ValueError(f"Missing {m} in {stats_path}")
+
+        return stat
+
+    stats_calc = StatsCalculator(req_dataset_meas)
+
+    for sample in tqdm(
+        dataset, total=dataset_length, descr="precomputing dataset stats", unit="sample"
+    ):
+        stats_calc.update(sample)
+
+    stat = stats_calc.finalize()
+    _ = stats_path.write_bytes(stat_adapter.dump_json(stat))
+
+    return stat
+
+
+class PredictCmd(CmdBase, WithSource):
+    """bioimageio-predict - Run inference on your data with a bioimage.io model."""
+
+    inputs: Union[str, Sequence[str]] = "model_inputs/*/{tensor_id}.*"
+    """model inputs
+
+    Either a single path/glob pattern including `{tensor_id}` to be used for all model inputs,
+    or a list of paths/glob patterns for each model input respectively.
+
+    For models with a single input a single path/glob pattern with `{tensor_id}` is also accepted.
+
+    `.npy` and any file extension supported by imageio
+    (listed at https://imageio.readthedocs.io/en/stable/formats/index.html#all-formats)
+    are supported.
+    """
+
+    outputs: Union[str, Sequence[str]] = (
+        "outputs_{model_id}/{sample_id}/{tensor_id}.npy"
+    )
+    """output paths analog to `inputs`"""
+
+    overwrite: bool = False
+    """allow overwriting existing output files"""
+
+    blockwise: bool = False
+    """process inputs blockwise"""
+
+    stats: Path = Path("model_inputs/dataset_statistics.json")
+    """path to dataset statistics
+    (will be written if it does not exist,
+    but the model requires statistical dataset measures)"""
+
+    def run(self):
+        model_descr = ensure_description_is_model(self.descr)
+
+        input_ids = [
+            t.name if isinstance(t, v0_4.InputTensorDescr) else t.id
+            for t in model_descr.inputs
+        ]
+        output_ids = [
+            t.name if isinstance(t, v0_4.OutputTensorDescr) else t.id
+            for t in model_descr.outputs
+        ]
+
+        glob_matched_inputs: Dict[str, List[Path]] = {}
+        n_glob_matches: Dict[int, List[str]] = {}
+
+        if isinstance(self.inputs, str):
+            if len(input_ids) > 1 and "{tensor_id}" not in self.inputs:
+                raise ValueError(
+                    f"{self.descr_id} needs inputs {input_ids}. Include '{{tensor_id}}' in `inputs` or provide multiple input paths/glob patterns."
+                )
+
+            inputs = [self.inputs.replace("{tensor_id}", t) for t in input_ids]
+        else:
```

Comment:
Not very important, but usually with if else statements, I try to make it more assertive

```python
if isinstance(self.inputs, str):
...
elif isinstance(self.inputs, list):
...
else:
    raise ValueError(f"Not expected type {self.inputs}")
```